cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryModule.java

```diff
@@ -93,20 +93,23 @@ public void process(Query query) {
         }
 
         if (config.slowQueryReportEnabled()) {
-            final Map<String, String> tags = new HashMap<>(4);
-            tags.put("host", hostname);
-            tags.put("statementType", query.statementType().toString());
-
-            final Map<String, String> fields = new HashMap<>(4);
-            fields.put("client", query.clientAddress());
-            fields.put("statement", query.statement());
-
-            final Measurement measurement = Measurement
-                    .create(service, (double) query.executionTimeInMilliseconds(), query.startTimeInMilliseconds(),
-                            TimeUnit.MILLISECONDS, tags, fields);
-
-            logger.trace("Measurement transformed: {}", measurement);
-            report(measurement);
+            if (config.queryTypesToLog().contains("ALL")
```

Comment:
There is [SlowQueryLogDecider](https://github.com/smartcat-labs/cassandra-diagnostics/blob/eaffb0f6b8ed457b616256a123b71b128d36f22d/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java) which we use to check if correct table is configured, if correct threshold and so on. It is better place to do it. We should maybe create a ticket to discuss if that is best o before any processing but that is outside of scope. Let us first put everything in one place.